HW1/P2/P2a.py

```diff
@@ -1,3 +1,24 @@
 from P2 import *
 
-# Your code here
+from pyspark import SparkContext
+import time
+#
+sc = SparkContext(appName="P2", pyFiles=[])
+
+rdd = sc.parallelize(xrange(2000))
+rdd = rdd.cartesian(rdd)
+rdd = rdd.partitionBy(100).map( lambda (k,v): ( (k,v), mandelbrot(v/500.0 - 2.,k/500.0 - 2.) ) )
+
+draw_image( rdd )
+
+start_t = time.time()
+effort = sum_values_for_partitions( rdd ).collect()
+print 'Time =', time.time() - start_t
+print effort
+
+print effort
+plt.figure()
+plt.xlabel("Iterations")
+plt.ylabel("Number of partitions")
+plt.hist(effort)
+plt.show()
```

Comment:
Hmmm interesting, yeah it's probably to do with the partitioning call I make after doing the cartesian. I've seen people paralellize with 10 partitions and then after the cartesian function will have 100 partitions. I've just gone with the default partitioning and then made 100 partitions after that so that might explain the difference.